sdks/go/pkg/beam/core/runtime/exec/sdf_invokers.go

```diff
@@ -0,0 +1,302 @@
+		return nil, errors.WithContext(err, "sdf RestrictionSize invoker")
+	}
+	return n, nil
+}
+
+func (n *rsInvoker) initCallFn() error {
+	// Expects a signature of the form:
+	// (key?, value, restriction) float64
+	// TODO(BEAM-9643): Link to full documentation.
+	switch fnT := n.fn.Fn.(type) {
+	case reflectx.Func2x1:
+		n.call = func(elms *FullValue, rest interface{}) float64 {
+			return fnT.Call2x1(elms.Elm, rest).(float64)
+		}
+	case reflectx.Func3x1:
+		n.call = func(elms *FullValue, rest interface{}) float64 {
+			return fnT.Call3x1(elms.Elm, elms.Elm2, rest).(float64)
+		}
+	default:
+		switch len(n.fn.Param) {
+		case 2:
+			n.call = func(elms *FullValue, rest interface{}) float64 {
+				n.args[0] = elms.Elm
+				n.args[1] = rest
+				return n.fn.Fn.Call(n.args)[0].(float64)
+			}
+		case 3:
+			n.call = func(elms *FullValue, rest interface{}) float64 {
+				n.args[0] = elms.Elm
+				n.args[1] = elms.Elm2
+				n.args[2] = rest
+				return n.fn.Fn.Call(n.args)[0].(float64)
+			}
+		default:
+			return errors.Errorf("RestrictionSize fn %v has unexpected number of parameters: %v",
+				n.fn.Fn.Name(), len(n.fn.Param))
+		}
+	}
+	return nil
+}
+
+// Invoke calls RestrictionSize given a FullValue containing an element and
+// the associated restriction, and returns a size.
+func (n *rsInvoker) Invoke(elms *FullValue, rest interface{}) (size float64) {
+	return n.call(elms, rest)
+}
+
+// Reset zeroes argument entries in the cached slice to allow values to be
+// garbage collected after the bundle ends.
+func (n *rsInvoker) Reset() {
+	for i := range n.args {
+		n.args[i] = nil
+	}
+}
+
+// ctInvoker is an invoker for CreateTracker.
+type ctInvoker struct {
+	fn   *funcx.Fn
+	args []interface{} // Cache to avoid allocating new slices per-element.
+	call func(rest interface{}) (tracker sdf.RTracker)
+}
+
+func newCreateTrackerInvoker(fn *funcx.Fn) (*ctInvoker, error) {
+	n := &ctInvoker{
+		fn:   fn,
+		args: make([]interface{}, len(fn.Param)),
+	}
+	if err := n.initCallFn(); err != nil {
+		return nil, errors.WithContext(err, "sdf CreateTracker invoker")
+	}
+	return n, nil
+}
+
+func (n *ctInvoker) initCallFn() error {
+	// Expects a signature of the form:
+	// (restriction) sdf.RTracker
+	// TODO(BEAM-9643): Link to full documentation.
+	switch fnT := n.fn.Fn.(type) {
+	case reflectx.Func1x1:
+		n.call = func(rest interface{}) sdf.RTracker {
+			return fnT.Call1x1(rest).(sdf.RTracker)
+		}
+	default:
+		if len(n.fn.Param) != 1 {
+			return errors.Errorf("CreateTracker fn %v has unexpected number of parameters: %v",
+				n.fn.Fn.Name(), len(n.fn.Param))
+		}
+		n.call = func(rest interface{}) sdf.RTracker {
+			n.args[0] = rest
+			return n.fn.Fn.Call(n.args)[0].(sdf.RTracker)
+		}
+	}
+	return nil
+}
+
+// Invoke calls CreateTracker given a restriction and returns an sdf.RTracker.
+// the associated restriction, and returns a size.
+func (n *ctInvoker) Invoke(rest interface{}) (tracker sdf.RTracker) {
```

Comment:
As a rule, naming return values where the meaning is already clear, and you aren't needing to modify the return in a defer is considered an anti pattern, unless it's a primitive type like string or a number, where the meaning of the type isn't necessarily clear.

It's also not necessary to do just because other functions/methods in the same file are doing it. There's no consistency requirement for it, and it can make things more confusing.

In short, drop the tracker, it's cleaner. :)

Reply:
Done.